src/s390_ecc.c

```diff
@@ -1743,16 +1743,19 @@ struct {				\
 #ifdef ICA_FIPS
 			if (fips & ICA_FIPS_MODE) {
 				fc |= 0x80; /* deterministic signature */
-				RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+				do {
+					RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+					rc = s390_kdsa(fc, param.buff, NULL, 0);
+				} while (rc);
+			} else {
+				rc = s390_kdsa(fc, param.buff, NULL, 0);
 			}
 #endif
 			rc = s390_kdsa(fc, param.buff, NULL, 0);
```

Comment:
still not correct. you still execute s390_kdsa in line 1754 when you come out of the if/else block.

I guess the s390_kdsa in line 1754 must go into an #else .... #endif of the #ifdef ICA_FIPS block.